src/Workspaces/MSBuildTest/NewlyCreatedProjectsFromDotNetNew.cs

```diff
@@ -0,0 +1,264 @@
+
+            CreateNewProject(templateName, projectDirectory, languageName, TestOutputHelper);
+
+            await AssertProjectLoadsCleanlyAsync(projectFilePath, ignoredDiagnostics ?? []);
+
+            return;
+
+            static string GetProjectFilePath(string projectDirectory, string languageName)
+            {
+                var projectName = new DirectoryInfo(projectDirectory).Name;
+                var projectExtension = languageName switch
+                {
+                    LanguageNames.CSharp => "csproj",
+                    LanguageNames.VisualBasic => "vbproj",
+                    _ => throw new ArgumentOutOfRangeException(nameof(languageName), actualValue: languageName, message: "Only C# and VB.NET projects are supported.")
+                };
+                return Path.Combine(projectDirectory, $"{projectName}.{projectExtension}");
+            }
+
+            static void CreateNewProject(string templateName, string outputDirectory, string languageName, ITestOutputHelper output)
+            {
+                var language = languageName switch
+                {
+                    LanguageNames.CSharp => "C#",
+                    LanguageNames.VisualBasic => "VB",
+                    _ => throw new ArgumentOutOfRangeException(nameof(languageName), actualValue: languageName, message: "Only C# and VB.NET projects are supported.")
+                };
+
+                TryCopyGlobalJson(outputDirectory);
+
+                var newResult = RunDotNet($"new \"{templateName}\" -o \"{outputDirectory}\" --language \"{language}\"", output, outputDirectory);
+
+                // Most templates invoke restore as a post-creation action. However, some, like the
+                // Maui templates, do not run restore since they require additional workloads to be
+                // installed.
+                if (newResult.Output.Contains("Restoring"))
+                {
+                    return;
+                }
+
+                try
+                {
+                    // Attempt a restore and see if we are instructed to install additional workloads.
+                    var restoreResult = RunDotNet($"restore", output, outputDirectory);
+                }
+                catch (InvalidOperationException ex) when (ex.Message.Contains("command: dotnet workload restore"))
+                {
+                    throw new InvalidOperationException($"The '{templateName}' template requires additional dotnet workloads to be installed. It should be excluded during template discovery. " + ex.Message);
+                }
+            }
+
+            static void TryCopyGlobalJson(string outputDirectory)
+            {
+                var tempGlobalJsonPath = Path.Combine(outputDirectory, "global.json");
+                try
+                {
+                    File.Copy(s_globalJsonPath, tempGlobalJsonPath);
+                }
+                catch (FileNotFoundException)
+                {
+                }
```

Comment:
I'm OK if you want to use File.Exists, and I won't tell @jaredpar.

Reply:
How do you know what I was thinking when I wrote this code? 

Will update to check that the file exists when we are locating it.